lib/statsd.js

```diff
@@ -402,8 +348,8 @@ Client.prototype.sendMessage = function (message, callback) {
     return;
   }
 
-  if (this.protocol === TCP && message.lastIndexOf('\n') !== message.length - 1) {
-    message += '\n';
+  if (!this.socket) {
+    return callback(new Error('No transport set. Check previous errors for details.'));
```

Comment:
```suggestion
    return callback(new Error('No protocol set. Check previous errors for details.'));
```

Since protocol is still the name exposed to users